src/controllers/mainController.ts

```diff
@@ -1783,16 +1784,54 @@ export default class MainController implements vscode.Disposable {
             // Untitled file was saved and connection will be transfered
             await this._connectionMgr.transferFileConnection(closedDocumentUri, this._lastSavedUri);
 
+            this._outputContentProvider.updateQueryRunnerUri(closedDocumentUri, this._lastSavedUri);
+            this._outputContentProvider.onUntitledFileSaved(closedDocumentUri, this._lastSavedUri);
+
+            let state = this._queryResultWebviewController.getQueryResultState(closedDocumentUri);
+
+            if (state) {
+                state.uri = this._lastSavedUri;
+                this._queryResultWebviewController.postNotification(
+                    DefaultWebviewNotifications.updateState,
+                    state,
+                );
+                this._queryResultWebviewController.setQueryResultState(this._lastSavedUri, state);
+                this._queryResultWebviewController.deleteQueryResultState(closedDocumentUri);
+            }
+            this._outputContentProvider.onUntitledFileSaved(closedDocumentUri, this._lastSavedUri);
```

Comment:
Looks like this logic is mostly duplicated below (only difference is the new URI).  Can you move it to a shared function?